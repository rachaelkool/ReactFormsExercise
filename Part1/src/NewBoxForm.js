import React, { useState } from "react";
import uuid from "uuid/v4";

function NewBoxForm({ addBox }) {
    const initialState = {
        height: '',
        width: '',
        backgroundColor: ''
    }

    const [formData, setFormData] = useState(initialState)

    const handleChange = e => {
        const {name, value} = e.target
        setFormData(data => ({
            ...data, 
            [name]: value
        }))
    }

    const handleSubmit = e => {
        e.preventDefault();
        addBox({ ...formData, id: uuid() });
        setFormData(initialState);
    };

    return (
        <div>
        <form onSubmit={handleSubmit}>
            <label htmlFor="height">Height:</label>
            <input
                id="height"
                type="text"
                name="height"
                placeholder="height"
                value={formData.height}
                onChange={handleChange}
            />
           
            <label htmlFor="width">Width:</label>
            <input
                id="width"
                type="text"
                name="width"
                placeholder="width"
                value={formData.width}
                onChange={handleChange}
            />
        
            <div>
            <label htmlFor="backgroundColor">Background Color:</label>
            <input
                id="backgroundColor"
                type="text"
                name="backgroundColor"
                placeholder="background color"
                value={formData.backgroundColor}
                onChange={handleChange}
            />
            </div>
            <button>Add a new box!</button>
        </form>
        </div>
    );
    }

export default NewBoxForm;