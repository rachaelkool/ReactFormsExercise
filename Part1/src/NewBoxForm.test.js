import React from "react";
import { render, fireEvent } from "@testing-library/react";
import NewBoxForm from "./NewBoxForm";

it('renders without crashing', function() {
    render(<NewBoxForm />);
});

it('matches snapshot', function() {
    const { asFragment } = render(<NewBoxForm />);
    expect(asFragment()).toMatchSnapshot();
});

it('runs the add function on form submit', function() {
    const createMock = jest.fn();
    const { getByText } = render(<NewBoxForm addBox={createMock} />);
    const createButton = getByText("Add a new box!");
    fireEvent.click(createButton);
    expect(createMock).toHaveBeenCalled();
});