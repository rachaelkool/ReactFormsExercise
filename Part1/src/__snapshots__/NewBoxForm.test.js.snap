// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="height"
      >
        Height:
      </label>
      <input
        id="height"
        name="height"
        placeholder="height"
        type="text"
        value=""
      />
      <label
        for="width"
      >
        Width:
      </label>
      <input
        id="width"
        name="width"
        placeholder="width"
        type="text"
        value=""
      />
      <div>
        <label
          for="backgroundColor"
        >
          Background Color:
        </label>
        <input
          id="backgroundColor"
          name="backgroundColor"
          placeholder="background color"
          type="text"
          value=""
        />
      </div>
      <button>
        Add a new box!
      </button>
    </form>
  </div>
</DocumentFragment>
`;
